### PR TITLE
Fix: Respect categorical option value 0 across node drawer and filter rules

### PR DESCRIPTION
**Pull request by buckhalt**
This PR addresses several bugs related to categorical bin variables with falsy values ( `0` is configurable in Architect web and desktop):

**Bug 1: CategoricalBin node drawer**
On CategoricalBin stages, dropping a node into a bin with a falsy `value` ( `0`) did not remove it from the uncategorized drawer. Dragging a node from another bin into that bin made the node reappear in the drawer. This was due to using `!!attributes[variable]` for the uncategorized nodes filter.

Fix: Switch to` isNil()` for the filter, consistent with approach in Ordinal Bins. Refactor to extract into a helper for easier testing, and add unit tests to cover regressions.

**Bug 2: Filter rules**
The same kind of bug was in the filter predicate. INCLUDES and EXCLUDES used `!value`, so a rule like INCLUDES `0`, where `0` is a value of a categorical variable, excluded every node which had the stored attribute as `0`.

Fix: Switch to `isNil()` for filters, and add unit tests.

**Bug 3: Export**
Exported data where a categorical attribute's value was `0` or `false` showed the option column as `FALSE` in exported data. Same issue,`isCategoricalOptionSelected` used `!attributeData. Only impacts single value categorical variables (categorical bins). 

Fix: Fix: Switch to `isNil()` in `isCategoricalOptionSelected`, and add unit tests covering stored `0` / `[0]` / `null`.

### CODE REVIEW

File: lib/interviewer/Interfaces/CategoricalBin/useCategoricalBins.ts [hunk 1]
```diff
@@ -33,6 +33,21 @@ type CategoricalBinPrompts = Extract<
...
+  otherVariable: string | undefined,
+) {
+  const activeVarExists = activePromptVariable
+    ? attributes[activePromptVariable] != null
+    : false;
+  const otherVarExists = otherVariable
+    ? attributes[otherVariable] != null
+    : false;
+
+  return !activeVarExists && !otherVarExists;
```
[the remark this answers is not in the document]
Comment (buckhalt) [reply lib/interviewer/Interfaces/CategoricalBin/useCategoricalBins.ts:59]: resolved [23b9a60](https://github.com/complexdatacollective/Fresco/pull/717/commits/23b9a609247a5a06dd05dd3200f352041537a089)

File: lib/interviewer/Interfaces/CategoricalBin/useCategoricalBins.ts [hunk 2]
```diff
@@ -53,17 +68,14 @@ export function useCategoricalBins() {
...
-    const attributes = node[entityAttributesProperty];
-
-    const activeVarExists = activePromptVariable
-      ? !!attributes[activePromptVariable]
-      : false;
-    const otherVarExists = otherVariable ? !!attributes[otherVariable] : false;
-
-    return !activeVarExists && !otherVarExists;
-  });
+  // Calcualte uncategorised nodes by filtering stageNodes to those that don't have a value for either the active prompt variable or the other variable
```
[the remark this answers is not in the document]
Comment (buckhalt) [reply lib/interviewer/Interfaces/CategoricalBin/useCategoricalBins.ts]: resolved [23b9a60](https://github.com/complexdatacollective/Fresco/pull/717/commits/23b9a609247a5a06dd05dd3200f352041537a089)

**Comment by jthrilly** [comment]
Aren't categorical option values (practically speaking) always strings? I know the type technically allows bools and numbers, but doesn't architect always produce strings? There's probably a case for narrowing the type rather than changing this.

**Comment by buckhalt** [comment]
> Aren't categorical option values (practically speaking) always strings? I know the type technically allows bools and numbers, but doesn't architect always produce strings? There's probably a case for narrowing the type rather than changing this.

Architect desktop & web do produce numeric options for categorical bins. They're both coercing integer inputs to numbers. They're not able to create bools in practice though so I dropped the tests for that specifically.

This wasn't an issue before in Fresco because we were previously wrapping the bin value in an array in the drop handler in `CategoricalListItem`
